src/hotspot/cpu/x86/x86_32.ad

```diff
@@ -12904,7 +12904,7 @@ instruct cmovPP_reg_LTGE(cmpOp cmp, flagsReg_long_LTGE flags, eRegP dst, eRegP s
 
 // Compare 2 longs and CMOVE doubles
 instruct cmovDDPR_reg_LTGE(cmpOp cmp, flagsReg_long_LTGE flags, regDPR dst, regDPR src) %{
-  predicate( UseSSE<=1 && _kids[0]->_kids[0]->_leaf->as_Bool()->_test._test == BoolTest::lt || _kids[0]->_kids[0]->_leaf->as_Bool()->_test._test == BoolTest::ge );
+  predicate( UseSSE<=1 && ( _kids[0]->_kids[0]->_leaf->as_Bool()->_test._test == BoolTest::lt || _kids[0]->_kids[0]->_leaf->as_Bool()->_test._test == BoolTest::ge ) );
   match(Set dst (CMoveD (Binary cmp flags) (Binary dst src)));
```

Comment:
@klaus4: After a closer review, this change and the following similar changes in this file do appear to be fixing bugs with these functions. It appears your changes match what the author was intending. Would you be willing to raise this as a question on the hotspot-dev list? 

https://mail.openjdk.java.net/mailman/listinfo/hotspot-dev